frontend/controller/service-worker.js

```diff
@@ -51,6 +51,34 @@ sbp('sbp/selectors/register', {
       await swRegistration.update()
       setInterval(() => sbp('service-worker/update'), HOURS_MILLIS)
 
+      // Send a 'ready' message to the SW and wait back for a response
+      // This way we ensure that Chelonia has been set up
+      await new Promise((resolve, reject) => {
+        const messageChannel = new MessageChannel()
+        messageChannel.port1.onmessage = (event) => {
+          if (event.data.type === 'ready') {
+            resolve()
+          } else {
+            reject(event.data.error)
+          }
+          messageChannel.port1.close()
+        }
+        messageChannel.port1.onmessageerror = () => {
+          reject(new Error('Message error'))
+          messageChannel.port1.close()
+        }
+
+        navigator.serviceWorker.ready.then((worker) => {
+          worker.active.postMessage({
+            type: 'ready',
+            port: messageChannel.port2
+          }, [messageChannel.port2])
+        }).catch((e) => {
+          reject(e)
+          messageChannel.port1.close()
+        })
+      })
+
```

Comment:
This seems like boilerplate we could abstract away in a function somewhere.

Can you create a function so that chunk is shortened like so?

```js
postWorkerMessage(worker, { type: 'ready' })
```

And if there are any other parts of the app that also do something like this, they can reuse this function.

Reply:
I could make it into another function, but this particular message is unique in how it's sent, that it expects a response and that it expects a response (`sbp` also expects a response, but is handled by a more general-purpose function)

Reply:
It's also not entirely general purpose because the handler for `onmessage` is custom